### DISCUSSION TITLE
pylint

### DISCUSSION THREAD

**Discussion by alexandrecuer** [pymodbus-dev/pymodbus#1165] [General]
@janiversen : as far as i understand, you are not using a .pylintrc file

Does the code of the library always comply to all of the pylint standards such as do not have too-many-arguments or too-many-instance-attributes ? 

I do not see disable comments in front of methods using many instance attributes for example, and so i was wondering how you manage pylint not to be verbose too much on those cases, without a .pylintrc file or a specific # disable comment ?

Best

**Comment by janiversen** [comment]
look in setup.cfg and tox.ini which are alternatives to .pylintrc

The code does NOT adhere to ALL pylint standards, as this would be contra-productive, setup.cfg contains a list of "not-wanted options".

The "too many attributes" as well as "too many arguments" are active, again see setup.cfg.

I try not to have a lot of .xxxxrc files but consolidate all in setup.cfg. tox.ini contains the way we invoke the tests (in ci, or command line with tox -e pylint etc.).

What we do need is someone to activate some of the pylint options which we want but we (more correctly I) did not have enough bandwidth to do, so please feel free to jump in, I am also happy to help you e.g. activating the missing pylint options (NOT the not wanted ones).
